[PLATFORM-403] Profile and Create Stream Page styling/updates

fonty1 opened the pull request:
Profile page and Create Stream pages updates, mostly involving styling/responsiveness.

This is mostly css to meet design spec, so to speed things up you guys could let Matt inspect the visual side of things, though please report anything major. 

Some changes to the logic inside the Create Stream Page, the main review item: (StreamPage/Show/ConfigureView/index.jsx)

**Functionality changes:**
- Stream config field items now get appended with an ID so that they can be reordered/live-edited without bugs. IDs get removed before POST/PUTing the stream.
- Stopped duplicate name possibility on live edit
- Ability to hide TOC section/heading via prop 'customStyled'
- Moved historical stuff into the historical component

**Not included:**
- Avatar stuff (designs/backed is a WIP)

juhah commented:
I'll just list here all that I come across that seems off 😄 

Stream list:

- Menu disappears (also sort dropdown is hugging the toolbar links):
<img width="845" alt="Screen Shot 2019-03-12 at 15 34 27" src="https://user-images.githubusercontent.com/1064982/54204200-66d15700-44dc-11e9-9709-a09793d840ca.png">

Stream detail page:

- Copy stream id button alignment:
<img width="909" alt="Screen Shot 2019-03-12 at 15 27 23" src="https://user-images.githubusercontent.com/1064982/54203849-b400f900-44db-11e9-9435-a61b306def43.png">
<img width="645" alt="Screen Shot 2019-03-12 at 15 26 50" src="https://user-images.githubusercontent.com/1064982/54203878-c11de800-44db-11e9-8e24-cc826c3f66f5.png">
<img width="696" alt="Screen Shot 2019-03-12 at 15 27 05" src="https://user-images.githubusercontent.com/1064982/54203891-cbd87d00-44db-11e9-9e3c-5aa59c831476.png">

- Delete field button gets lost under the side when hovering:
<img width="966" alt="Screen Shot 2019-03-12 at 15 27 47" src="https://user-images.githubusercontent.com/1064982/54203911-d3982180-44db-11e9-9272-c7151c23828c.png">

- API access type goes to next line:
<img width="562" alt="Screen Shot 2019-03-12 at 15 36 46" src="https://user-images.githubusercontent.com/1064982/54204330-b57ef100-44dc-11e9-8f9d-75a8c8fb1888.png">

- API access type dropdown goes to next line (edit):
<img width="574" alt="Screen Shot 2019-03-12 at 15 28 20" src="https://user-images.githubusercontent.com/1064982/54203962-f0345980-44db-11e9-9a20-a5368b113859.png">

- (Very) slight difference in cancel button background colour:
<img width="245" alt="Screen Shot 2019-03-12 at 15 29 09" src="https://user-images.githubusercontent.com/1064982/54204070-2bcf2380-44dc-11e9-9593-ea0dcd888a9f.png">

Profile page:

- Very thin menu in tablet resolution:
<img width="905" alt="Screen Shot 2019-03-12 at 15 33 32" src="https://user-images.githubusercontent.com/1064982/54204115-430e1100-44dc-11e9-8737-0203c1257f6b.png">

fonty1 commented:
thanks @juhah 

> Menu disappears (also sort dropdown is hugging the toolbar links)

Out of scope for this PR, I will fix this in the Userpages styling branch.

> Copy stream id button alignment (ID prefill)

Thanks, fixed.

> Delete field button gets lost under the side when hovering

Fixed by trimming the tabletish area margins

> API access type goes to next line
> API access type dropdown goes to next line (edit)

Correct, this was meant to be hidden on tablet down. I will improve this further once we have backend editing function of those api key permissions.

> (Very) slight difference in cancel button background colour (keyname / add)

Transparency style added here.

> Very thin menu in tablet resolution

Good catch, this real estate utilization is poor in the tablet range. I've applied a fix here, please have another look.

I also changed 'map' -> 'object' field type terminology (Henri's request). And a bunch of other smaller details that I missed on the first round. Please take another look and let me know how you go.

fonty1 commented:
> Looks a lot better now! I noticed that adding API keys for stream does not work when creating a new stream. After saving it works as expected. Maybe not part of this PR but we probably should disable the add button when stream does not yet have id.

Interesting, good catch. I'll see what can be done here.

juhah commented:
Looks very good, all issues fixed except the minor background difference in the cancel button.

<img width="334" alt="Screen Shot 2019-03-18 at 13 15 34" src="https://user-images.githubusercontent.com/1064982/54526644-0af54b00-4980-11e9-9f79-a6a247050010.png">

In profile page (tablet resolution), the add key form background extends to the edge which looks a bit awkward:
<img width="902" alt="Screen Shot 2019-03-18 at 13 14 25" src="https://user-images.githubusercontent.com/1064982/54526677-252f2900-4980-11e9-91b5-8418ab00a9ce.png">

fonty1 commented:
> Looks very good, all issues fixed except the minor background difference in the cancel button.

> In profile page (tablet resolution), the add key form background extends to the edge which looks a bit awkward:

Those should be fixed - assuming you mean the hover state for the cancel button? That cancel button does follow the style guide but the style guide isn't prepared for this edge case of a light grey container. I've removed it for now which looks a bit better.

fonty1 commented on app/src/userpages/components/ProfilePage/IntegrationKeyHandler/index.jsx:
```diff
@@ -52,9 +50,14 @@ export class IntegrationKeyHandler extends Component<Props> {
                 <AddKeyField
-                    label={I18n.t('userpages.profilePage.ethereumAddress.addNewAddress')}
+                    // $FlowFixMe
+                    label={this.props.integrationKeys[0] ?
```
thanks @timoxley, good catch and I appreciate the code snippet 👏 👌

I will make sure I circle back to these sorts of fixes in the future before merge.
